sdk/eventhub/Azure.Messaging.EventHubs/tests/Amqp/AmqpMessageConverterTests.cs

```diff
@@ -710,5 +713,323 @@ public void CreateBatchFromMessagesWithMultipleEventsAssignsThePartitionKeyToBod
+            using var message = AmqpMessage.Create(bodyStream, true);
+
+            var typeDescriptor = (AmqpSymbol)"INVALID";
+            var describedProperty = new DescribedType(typeDescriptor, 1234);
+            message.ApplicationProperties.Map.Add(typeDescriptor.ToString(), describedProperty);
+
+            var converter = new AmqpMessageConverter();
+            var eventData = converter.CreateEventFromMessage(message);
+
+            Assert.That(eventData, Is.Not.Null, "The event should have been created.");
+            Assert.That(eventData.Body, Is.Not.Null, "The event should a body.");
+            Assert.That(eventData.Properties.Any(), Is.False, "The event should not have a set of application properties.");
+
+            var containsValue = eventData.Properties.TryGetValue(typeDescriptor.ToString(), out var _);
+            Assert.That(containsValue, Is.False, $"The event properties should not contain the described property.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="AmqpMessageConverter.CreateEventFromMessage" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void CreateEventFromMessagePopulatesSystemProperties()
+        {
+            var offset = 123;
+            var sequenceNumber = (Int64.MaxValue - 10);
+            var enqueueTime = DateTimeOffset.Parse("2015-10-27T12:00:00Z");
+            var partitionKey = "OMG! partition!";
+
+            using var bodyStream = new MemoryStream(new byte[] { 0x11, 0x22, 0x33 }, false);
+            using var message = AmqpMessage.Create(bodyStream, true);
+
+            message.ApplicationProperties.Map.Add("First", 1);
+            message.ApplicationProperties.Map.Add("Second", "2");
+
+            message.MessageAnnotations.Map.Add(AmqpProperty.Offset, offset.ToString());
+            message.MessageAnnotations.Map.Add(AmqpProperty.SequenceNumber, sequenceNumber);
+            message.MessageAnnotations.Map.Add(AmqpProperty.EnqueuedTime, new DescribedType(AmqpProperty.Descriptor.DateTimeOffset, enqueueTime.Ticks));
+            message.MessageAnnotations.Map.Add(AmqpProperty.PartitionKey, partitionKey);
+
+            var converter = new AmqpMessageConverter();
+            var eventData = converter.CreateEventFromMessage(message);
+
+            Assert.That(eventData, Is.Not.Null, "The event should have been created.");
+            Assert.That(eventData.Body, Is.Not.Null, "The event should a body.");
+            Assert.That(eventData.Properties.Count, Is.EqualTo(message.ApplicationProperties.Map.Count()), "The event should have a set of properties.");
+            Assert.That(eventData.Offset, Is.EqualTo(offset), "The offset should match.");
+            Assert.That(eventData.SequenceNumber, Is.EqualTo(sequenceNumber), "The sequence number should match.");
+            Assert.That(eventData.EnqueuedTime, Is.EqualTo(enqueueTime), "The enqueue time should match.");
```

Comment:
enqueue => enqueued? (in the message as well).

Just to keep consistency with `eventData.EnqueuedTime`.